app/src/main/java/com/cornellappdev/android/volume/ui/viewmodels/FlyerUploadViewModel.kt

```diff
@@ -18,68 +22,53 @@ class FlyerUploadViewModel @Inject constructor
     (
     private val organizationsRepository: OrganizationRepository,
     private val flyersRepository: FlyerRepository,
+    private val networkApi: NetworkApi,
 ) : ViewModel() {
-    data class FlyerUploadUiState(
-        val uploadFlyerResult: ResponseState<String> = ResponseState.Loading,
-        val organizationInfoResult: ResponseState<Organization> = ResponseState.Loading,
-    )
+    private val _orgFlow: MutableStateFlow<ResponseState<Organization>> =
+        MutableStateFlow(ResponseState.Loading)
+    val orgFlow = _orgFlow.asStateFlow()
 
-    var uploadFlyerUiState by mutableStateOf(FlyerUploadUiState())
-        private set
+    private val _flyerFlow: MutableStateFlow<ResponseState<Flyer>> =
+        MutableStateFlow(ResponseState.Loading)
+    val flyerFlow = _flyerFlow.asStateFlow()
 
-    fun getOrganization(id: String) = viewModelScope.launch {
-        uploadFlyerUiState = try {
-            uploadFlyerUiState.copy(
-                organizationInfoResult = ResponseState.Success(
-                    organizationsRepository.getOrganizationById(
-                        id
-                    )!!
-                )
-            )
-        } catch (e: Exception) {
-            uploadFlyerUiState.copy(
-                organizationInfoResult = ResponseState.Error(listOf())
-            )
+    private val _uploadResult: MutableStateFlow<ResponseState<Flyer>> =
+        MutableStateFlow(ResponseState.Loading)
+    val uploadResultFlow = _uploadResult.asStateFlow()
+
+    fun initViewModel(organizationSlug: String, flyerId: String?) = viewModelScope.launch {
+        try {
+            _orgFlow.value =
+                ResponseState.Success(organizationsRepository.getOrganizationBySlug(organizationSlug)!!)
+        } catch (_: Exception) {
+            _orgFlow.value = ResponseState.Error()
+        }
+        try {
+            flyerId?.let {
+                _flyerFlow.value = ResponseState.Success(flyersRepository.fetchFlyerById(it))
+            }
+        } catch (_: Exception) {
+            _flyerFlow.value = ResponseState.Error()
         }
     }
```

Comment:
Oh neat, I didn't know Apollo allowed for that. That's definitely something I could experiment with over thanksgiving break, but for now I think this will suffice.